File: jats-demo-webapp/src/main/ts/model/ProductVariant.ts
class ProductVariant
{
	private uom : Uom;

	private mainProduct : Product;

	private requiredComponents : Component[];

	private optionalComponents : Component[];

	private id : number;

	public getUom() { return this.uom; }
	public setUom( uom : Uom ) { this.uom = uom; }

	public getMainProduct() { return this.mainProduct; }
	public setMainProduct( mainProduct : Product ) { this.mainProduct = mainProduct; }

	public getRequiredComponents() { return this.requiredComponents; }
	public setRequiredComponents( requiredComponents : Component[] ) { this.requiredComponents = requiredComponents; }

	public getOptionalComponents() { return this.optionalComponents; }
	public setOptionalComponents( optionalComponents : Component[] ) { this.optionalComponents = optionalComponents; }

	public getId() { return this.id; }
	public setId( id : number ) { this.id = id; }


}

File: jats-demo-model/src/main/java/com/github/pepe79/jats/demo/model/ProductVariant.java
package com.github.pepe79.jats.demo.model;

import java.util.List;

public class ProductVariant
{
	private int id;

	private Product mainProduct;

	private List<Component> requiredComponents;

	private List<Component> optionalComponents;

	private Uom uom;
	
	public Uom getUom()
	{
		return uom;
	}

	public void setUom(Uom uom)
	{
		this.uom = uom;
	}

	public Product getMainProduct()
	{
		return mainProduct;
	}

	public void setMainProduct(Product mainProduct)
	{
		this.mainProduct = mainProduct;
	}

	public List<Component> getRequiredComponents()
	{
		return requiredComponents;
	}

	public void setRequiredComponents(List<Component> requiredComponents)
	{
		this.requiredComponents = requiredComponents;
	}

	public List<Component> getOptionalComponents()
	{
		return optionalComponents;
	}

	public void setOptionalComponents(List<Component> optionalComponents)
	{
		this.optionalComponents = optionalComponents;
	}

	public int getId()
	{
		return id;
	}

	public void setId(int id)
	{
		this.id = id;
	}

}
